demisto_sdk/commands/content_graph/interface/neo4j/queries/validations.py

```diff
@@ -267,6 +267,28 @@ def validate_core_packs_dependencies(
     }
 
 
+def validate_hidden_pack_dependencies(
+    tx: Transaction,
+    pack_ids: List[str],
+):
+    query = f"""// Returns DEPENDS_ON relationships to packs which are hidden
+        MATCH (pack1)-[r:DEPENDS_ON{{mandatorily:true}}]->(pack2)
+        WHERE pack2.object_id IN {pack_ids} AND pack2.hidden
+        AND NOT r.is_test
+        and NOT pack1.hidden
+        and NOT pack1.deprecated
+        RETURN pack1, collect(r) as relationships, collect(pack2) as nodes_to
+        """
```

Comment:
```suggestion
    query = f"""// Returns DEPENDS_ON relationships to packs which are hidden
MATCH (pack1)-[r:{RelationshipType.DEPENDS_ON}{{mandatorily: true}}]->(pack2)
WHERE pack2.object_id IN {pack_ids} AND pack2.hidden
AND NOT r.is_test
and NOT pack1.hidden
and NOT pack1.deprecated
RETURN pack1, collect(r) as relationships, collect(pack2) as nodes_to"""
```

Reply:
also I think worth doing:
```
WHERE pack1.object_id IN {pack_ids} OR pack2.object_id IN {pack_ids}
AND pack2.hidden
...
```
to ensure we don't add a dependency to an already hidden pack. I think this will require a slight adjustment in `validate_hidden_packs_do_not_have_mandatory_dependencies`

Reply:
sounds good, however the validation shouldn't run **only** on pack IDs that were changed?
in this case we generalize it to run on every possible pack, no?